engine/clients/gsi/upload.py

```diff
@@ -65,37 +86,45 @@ def fvs_upload(cls):
             m = cls.upload_params["m"] or None
             efc = cls.upload_params["efConstruction"] or None
 
+        if DEBUG: print("GSIUploader: fvs upload: params=", nbits, search_type, top, m, efc)
+
         # convert dataset to float32...
         data = np.load(GSI_DEFAULT_DATA_PATH)
+        if DEBUG: print("GSIUploader: fvs_upload: dataset before", data.shape, data.dtype, data[0][0])
+
         if not isinstance(np.float32, type(data[0][0])):
             data = np.float32(data)
         if os.path.exists(GSI_DEFAULT_DATA_PATH):
             os.remove(GSI_DEFAULT_DATA_PATH)
         np.save(GSI_DEFAULT_DATA_PATH, data)
+        data = np.load(GSI_DEFAULT_DATA_PATH)
+        
+        if DEBUG: print("GSIUploader: fvs_upload: dataset after", data.shape, data.dtype, data[0][0])
 
         # import dataset
+        if DEBUG: print("GSIUploader: fvs_upload: dataset import parms", search_type, GSI_DEFAULT_DATA_PATH, nbits, m, efc)
         response = cls.client.datasets_apis.controllers_dataset_controller_import_dataset(
             ImportDatasetRequest(records=GSI_DEFAULT_DATA_PATH, search_type=search_type, train_ind=True,\
                                  nbits=nbits, dataset_name="QdrantBench", m_number_of_edges=m, ef_construction=efc),
             cls.client.allocation_id
         )
-        print("got datasetid=", response.dataset_id)
+        if DEBUG: print("GSIUploader: fvs_upload: got datasetid=", response.dataset_id)
         dataset_id = response.dataset_id
         cls.client.dataset_ids.append(dataset_id)
 
         # train status
         train_status = cls.client.datasets_apis.controllers_dataset_controller_get_dataset_status(
                 dataset_id=dataset_id, allocation_token=cls.client.allocation_id
             ).dataset_status
-        print("training, status currently:", train_status)
+        if DEBUG: print("GSUploader: fvs_upload: training, status currently:", train_status)
         while train_status != "completed":
             train_status = cls.client.datasets_apis.controllers_dataset_controller_get_dataset_status(
                 dataset_id=dataset_id, allocation_token=cls.client.allocation_id
             ).dataset_status
-            time.sleep(10)
```

Comment:
I set it to 10 because we don't care about FVS upload speed and the status requests were flooding the logs

Reply:
sounds good...and, oops...that was debug code :)